### DISCUSSION TITLE
Pydantic does not work as expected when table = true. Maybe need "pydantic first"?

### DISCUSSION THREAD

**Discussion by Xtreemrus** [fastapi/sqlmodel#561] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional

from sqlmodel import SQLModel, Field

class SongBase(SQLModel):
    name: str
    artist: str
    year: Optional[int] = None

class SongCreate(SongBase):
    pass

class Song(SongBase, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)


song = Song(id = 'not int value', name = 'some name', artist = 'some artist', year = 'other not int value')
print(song)
```

### Description

Surprisingly, the code from the example will work without errors.

As far as I understand, the purpose of SQLModel is to take the best from SQLAlchemy and Pydantic, but as a user it was not obvious to me that adding table = true completely disables pydantic validation. Similar to "mobile first" in web design, maybe we can think about "pydantic first" style development, where by default we expect that if there are no errors, all data is correct?

### Operating System

Windows

### Operating System Details

Windows 11

### SQLModel Version

0.0.8

### Python Version

3.9.9

**Comment by antont** [comment]
It will fail if you try to add it to the database. Seems somewhat nice also, that for DB classes, the DB validation is used.

This has been discussed in several issues too, though, and IIRC folks have modified sqlmodel to do pydantic validation there also.

**Reply by Duckling92** [reply]
Is it intended that validation is skipped when `table=true`? I would be grateful if you could point me to an explanation, as this seems as a peculiar choice to me (probably due to me lack of understanding), if it indeed is intended.

**Comment by YuriiMotov** [comment]
There is a related issue: https://github.com/fastapi/sqlmodel/issues/453
Let's track it there.

Also, old closed issue: https://github.com/fastapi/sqlmodel/issues/52
